### PR TITLE
Feat job: add option to create independent children

### PR DESCRIPTION
**Pull request by roggervalf**
There is a case where we only want to keep the relationship between parent-child and not waiting for children, so it is possible that parent completes first
I have also this case, I would like your opinion about it.
re #599

### CODE REVIEW

File: src/interfaces/flow-job.ts [hunk 1]
```diff
@@ -5,6 +5,6 @@ export interface FlowJob {
   queueName: string;
   data?: any;
   prefix?: string;
-  opts?: Omit<JobsOptions, 'parent'>;
+  opts?: Omit<JobsOptions, 'parent' | 'independence'>;
```
Comment (ccollie) [inline src/interfaces/flow-job.ts:8]: "independence" should probably be "independent"

Comment (roggervalf) [reply src/interfaces/flow-job.ts]: makes sense to me, thank you

**Comment by manast** [comment]
Trying to wrap my head around this. So what is the reason for having the "independents" jobs? can't this be solved on top of BullMQ in user code?

**Comment by roggervalf** [comment]
> Trying to wrap my head around this. So what is the reason for having the "independents" jobs? can't this be solved on top of BullMQ in user code?

Sure I can explain this idea, in my case I also have some children jobs that I don't need to wait to wait for them, but for easy tracking would be nice to have the relationship as those jobs are still children of one parent, I was thinking on extending the children get methods to support this case, for now I extended the getDependenciesCount

**Comment by Visot** [comment]
what about orphans instead of independents?

**Comment by roggervalf** [comment]
> what about orphans instead of independents?

I tempted to use this word now, ty

**Comment by manast** [comment]
> > what about orphans instead of independents?
> 
> I tempted to use this word now, ty

Would "orphan" (lack of parent) really represent the intent? (considering your explanation above where you wrote that the case is to keep track children of a parent but without needing to wait for them).

**Comment by roggervalf** [comment]
> > > what about orphans instead of independents?
> > 
> > 
> > I tempted to use this word now, ty
> 
> Would "orphan" (lack of parent) really represent the intent? (considering your explanation above where you wrote that the case is to keep track the children of a parent but without needing to wait for them).

Yes, that's the intention, would you prefer independents in this case?

**Comment by manast** [comment]
Sorry for the late response.  I am a bit reluctant for this kind of functionality that is for bookkeeping purposes if it can build on top of BullMQ, the reason is that this will otherwise lead to feature creep and I really want to avoid that at all costs.

**Comment by roggervalf** [comment]
> Sorry for the late response. I am a bit reluctant for this kind of functionality that is for bookkeeping purposes if it can build on top of BullMQ, the reason is that this will otherwise lead to feature creep and I really want to avoid that at all costs.

No worries, I can keep this pr opened to show the idea about how it could be implemented

**Pull request closed without merging**